libs/domain/push-notification/src/services/providers/web-push/web-push.ts

```diff
@@ -0,0 +1,80 @@
+import { from, map, Observable, of, switchMap } from 'rxjs';
+import { PushProvider } from '../push-provider.service';
+
+export interface WebPushProviderOptions {
+  /**
+   * Base64 encoded public VAPID Web Push key
+   *
+   * A keypair can be generated at https://vapidkeys.com/
+   */
+  applicationServerKey?: string;
+  /**
+   * Indicate if push message will always trigger visible notifications.
+   *
+   * By default it is set to `true` because Chrome requires it.
+   * @default true
+   * @see https://docs.google.com/document/d/13VxFdLJbMwxHrvnpDm8RXnU41W2ZlcP0mdWWe9zXQT8/edit
+   */
+  userVisibleOnly?: boolean;
+}
+
+export class WebPushProvider implements PushProvider<PushSubscriptionJSON> {
+  protected pushManager$ = from(navigator.serviceWorker.ready).pipe(
+    map((serviceWorker) => serviceWorker.pushManager)
+  );
+
+  constructor(protected options?: WebPushProviderOptions) {}
+
+  init(): Observable<void> {
+    return of(undefined);
+  }
+
+  getSubscription(): Observable<PushSubscriptionJSON> {
+    return this.getExistingSubscription().pipe(
+      switchMap((subscription) =>
+        subscription ? of(subscription) : this.createSubscription()
+      ),
+      map((subscription) => subscription.toJSON())
+    );
+  }
+
+  deleteSubscription(): Observable<boolean> {
+    return this.getExistingSubscription().pipe(
+      switchMap((subscription) => {
+        if (subscription) {
+          return subscription.unsubscribe();
```

Comment:
`subscription ? subscription.unsubscribe() : of(true)`

And what is the purpose of `of(true)`?